persist/boltdb_test.go

```diff
@@ -0,0 +1,167 @@
+package persist
+
+import (
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/bolt"
+)
+
+const numTestInputs = 25
+
+var dbVersions = []string{"0.0.0", "7.0.4", "asdf"}
+
+type testInput struct {
+	dbMetadata Metadata
+	dbFilename string
+}
+
+var testInputs = make([]testInput, numTestInputs)
+
+var testBuckets = [][]byte{
+	[]byte("FakeBucket"),
+	[]byte("FakeBucket123"),
+	[]byte("FakeBucket123!@#$"),
+	[]byte("Another Fake Bucket"),
+	[]byte("FakeBucket" + RandomSuffix()),
+	[]byte("_"),
+	[]byte(" asdf"),
+}
+
+// TestOpenDatabase tests calling OpenDatabase on the following types of
+// databases:
+// - a database that has not yet been created
+// - an existing empty database
+// - an existing nonempty database
+// Along the way, it also tests calling Close on:
+// - a newly-created database
+// - a newly-filled database
+// - a newly-emptied database
+func TestOpenDatabase(t *testing.T) {
+	// Create a folder for the database file. If a folder by that name exists
+	// already, it will be replaced by an empty folder.
+	testdir := build.TempDir(persistDir, "TestOpenNewDatabase")
+	err := os.MkdirAll(testdir, 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Generate ten random testInputs for building database files.
```

Comment:
better not to use a word like 'ten', because sometimes you change the variable but forget the constant :) (same problem you fixed above - probably better to remove the number altogether here too)